esp/esp/program/modules/handlers/volunteermanage.py

```diff
@@ -91,10 +98,55 @@ def volunteering(self, request, tl, one, two, module, extra, prog):
         else:
             form = VolunteerRequestForm(program=prog)
 
-        context['form'] = form
+        context['shift_form'] = form
+        if 'import_request_form' not in context:
+            context['import_request_form'] = VolunteerImportForm()
         context['requests'] = self.program.getVolunteerRequests()
         return render_to_response('program/modules/volunteermanage/main.html', request, context)
 
+    def volunteer_import(self, request, tl, one, two, module, extra, prog):
+        context = {}
+        response = None
+
+        import_form = VolunteerImportForm(request.POST)
+        if not import_form.is_valid():
+            context['import_request_form'] = import_form
+        else:
+            past_program = import_form.cleaned_data['program']
+            start_date = import_form.cleaned_data['start_date']
+            if past_program == prog:
+                context['import_error'] = "You can only import shifts from previous programs"
+            else:
+                #Figure out timeslot dates
+                prev_timeslots = []
+                prev_requests = past_program.getVolunteerRequests().order_by('timeslot__start')
+                for prev_request in prev_requests:
+                    prev_timeslots.append(prev_request.timeslot)
+                time_delta = start_date - prev_timeslots[0].start.date()
+                for i,orig_timeslot in enumerate(prev_timeslots):
+                    new_timeslot = Event.objects.get_or_create(
```

Comment:
nit: usually clearer to do `new_timeslot, _ = Event.objects.get_or_create(...)` rather than `new_timeslot = Event.object.get_or_create(...)[0]`.